core/primitives/src/shard_layout.rs

```diff
@@ -361,6 +361,10 @@ impl ShardUId {
     pub fn shard_id(&self) -> ShardId {
         ShardId::from(self.shard_id)
     }
+
+    pub fn global() -> Self {
+        Self { version: 0, shard_id: u32::MAX }
```

Comment:
Can you use the constant here? 
also the type mismatch may be already causing problems here